src/lib/pay-request/client.js

```diff
@@ -19,27 +21,48 @@ const serviceApiMethodUtils = require('./api_utils')
 
 const PAY_REQUEST_TIMEOUT = 10000
 
-const timestampRequest = function timestampRequest(request) {
+const configureRequest = function configureRequest(request) {
+  // @TODO(sfount) share definitions for session closure among modules
+  const session = getNamespace('govuk-pay-logging')
+
+  // if a correlation ID has been passed from reverse proxy -- pass it on to internal services
+  if (session.get('correlation_id')) {
+    request.headers['x-request-id'] = session.get('correlation_id')
+  }
   request.metadata = { start: new Date() }
   return request
 }
 
 const logSuccessfulResponse = function logSuccessfulResponse(response) {
+  const logContext = {
+    service: this.metadata.serviceKey,
+    method: response.request.method,
+    url: response.config.url,
+    duration: response.config.metadata.duration
+  }
+
   response.config.metadata.end = new Date()
   response.config.metadata.duration = response.config.metadata.end - response.config.metadata.start
-  logger.debug(`[${this.metadata.serviceKey}] "${response.request.method}" success from ${response.config.url} (${response.config.metadata.duration}ms)`)
+  logger.info('Pay request: internal service success fulfilled', logContext)
```

Comment:
[nothing major] `success fulfilled` sounds a bit odd to me.

Reply:
You're right, had hoped to rely on context object for this log line but will update the message to include the service and the code for looking from a distance